testing/internal/e2e/infra/docker.go

```diff
@@ -303,3 +303,44 @@ func StartOpenSshServer(t testing.TB, pool *dockertest.Pool, network *dockertest
 		UriNetwork: networkAlias,
 	}
 }
+
+// StartMysql starts a MySQL database in a docker container.
+// Returns information about the container
+func StartMysql(t testing.TB, pool *dockertest.Pool, network *dockertest.Network, repository, tag string) *Container {
+	t.Log("Starting MySQL database...")
+	c, err := LoadConfig()
+	require.NoError(t, err)
+
+	err = pool.Client.PullImage(docker.PullImageOptions{
+		Repository: fmt.Sprintf("%s/%s", c.DockerMirror, repository),
+		Tag:        tag,
+	}, docker.AuthConfiguration{})
+	require.NoError(t, err)
+
+	networkAlias := "e2emysql"
+	mysqlDb := "e2eboundarydb"
+	mysqlUser := "e2eboundary"
+	mysqlPassword := "e2eboundary"
+	mysqlRootPassword := "rootpassword"
+
+	resource, err := pool.RunWithOptions(&dockertest.RunOptions{
+		Repository: fmt.Sprintf("%s/%s", c.DockerMirror, repository),
+		Tag:        tag,
+		Env: []string{
+			"MYSQL_DATABASE=" + mysqlDb,
+			"MYSQL_USER=" + mysqlUser,
+			"MYSQL_PASSWORD=" + mysqlPassword,
+			"MYSQL_ROOT_PASSWORD=" + mysqlRootPassword,
+		},
+		ExposedPorts: []string{"3306/tcp"},
+		Name:         networkAlias,
+		Networks:     []*dockertest.Network{network},
+	})
+	require.NoError(t, err)
+
+	return &Container{
+		Resource:     resource,
+		UriLocalhost: "http://localhost:3306",
+		UriNetwork:   "http://e2emysql:3306",
```

Comment:
It's neat that we don't have to provide credentials to this function anymore, but this makes the container harder to use (the user has to guess to look at the environment variables). It looks like MySQL [supports using URI connection strings](https://dev.mysql.com/doc/refman/8.4/en/connecting-using-uri-or-key-value-pairs.html#connecting-using-uri) in the same way as postgres does. What do you think about putting the credentials in these returned strings instead and extract them out in the caller?
```suggestion
		UriLocalhost: fmt.Sprintf("mysql://%s:%s@localhost:3306/%s", mysqlUser, mysqlPassword, mysqlDb),
		UriNetwork:   fmt.Sprintf("mysql://%s:%s@%s:3306/%s", mysqlUser, mysqlPassword, networkAlias, mysqlDb),
```

In the test we can extract this using `url.Parse`:

```go
u, err := url.Parse(container.UriLocalhost)
if err != nil {
...
}
user := u.User.Username()
pw := u.User.Password()
```

This way it behaves just like the postgres function.